inductiva/_cli/pager.py

```diff
@@ -0,0 +1,312 @@
+
+        Always returns False.
+        """
+        return False
+
+    def seekable(self) -> bool:
+        """Return whether the object supports random access.
+
+        Always returns False.
+        """
+        return False
+
+    def fileno(self) -> int:
+        raise io.UnsupportedOperation("Underlying file object has no fileno")
+
+    def read(self, n: int = ...) -> AnyStr:
+        raise io.UnsupportedOperation("Underlying file object is not readable")
+
+    def readline(self, limit: int = ...) -> AnyStr:
+        raise io.UnsupportedOperation("Underlying file object is not readable")
+
+    def readlines(self, hint: int = ...) -> typing.List[AnyStr]:
+        raise io.UnsupportedOperation("Underlying file object is not readable")
+
+    def seek(self, offset: int, whence: int = ...) -> int:
+        # self._txtpad.move(offset, 0)
+        # self._current_line = offset
+        raise io.UnsupportedOperation("File-like object is not seekable")
+
+    def tell(self) -> int:
+        raise io.UnsupportedOperation(
+            "File-like object does not support the tell method")
+
+    def truncate(self, size: Optional[int] = ...) -> int:
+        # self._txtpad.clrtobot()
+        # self._line_count = self._current_line
+        raise io.UnsupportedOperation(
+            "File-like object does not support the truncate method")
+
+    def __next__(self) -> AnyStr:
+        raise TypeError("File-like object is not an iterator")
+
+    def __iter__(self) -> Iterator[AnyStr]:
+        raise TypeError("File-like object is not iterable")
+
+
+class Scheduler(threading.Thread):
+
+    def __init__(self, every: float, action, args=(), kwargs=None):
+        super().__init__()
+        self.every = every
+        self.action = action
+        self.args = args
+        self.kwargs = kwargs or {}
+
+        self._event = threading.Event()
+        self._count = 0
+
+    @property
+    def count(self):
+        """Return the number of times the action has been executed."""
+        return self._count
+
+    def run(self):
+        """Run the scheduler until the stop method is called."""
+        elapsed = 0.0
+        while not self._event.is_set():
+            self._count += 1
+
+            now = time.time()
+            self.action(*self.args, **self.kwargs)
+            elapsed = time.time() - now
+
+            # take into account the time the action takes to complete
+            if elapsed < self.every:
+                self._event.wait(self.every - elapsed)
+
+    def stop(self):
+        """Stop the scheduler."""
+        self._event.set()
+
+
+def main():
+    every = 5.0  # seconds
+
+    def producer(fout: PagedOutput, schedulr: Scheduler):
+        fout.clear()
+        for i in range(50):
+            fout.write(f"--- line {schedulr.count} {i+1} ---\n")
+        fout.scroll_to(0)
+
+    with PagedOutput(f"> every {every} s: inductiva tasks list -n 10") as f:
+        scheduler = Scheduler(every, producer, args=())
```

Comment:
A bit confused here... The `Scheduler` takes in the producer, which itself takes a Scheduler...

So, how is it actually initialized?